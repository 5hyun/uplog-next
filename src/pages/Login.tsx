import React, { FormEvent, useCallback } from 'react';
import { useMessage } from '../hooks/useMessage.ts';
import { MdOutlineMailOutline } from 'react-icons/md';
import useInput from '../hooks/useInput.ts';
import { AiOutlineLock } from 'react-icons/ai';
import { Link } from 'react-router-dom';

export default function Login() {
  const { showMessage, contextHolder } = useMessage();
  const [email, onChangeEmail, setEmail] = useInput('');
  const [password, onChangePassword, setPassword] = useInput('');

  const onSubmit = useCallback((e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (!email || !password) showMessage('warning', '이메일과 비밀번호를 입력해주세요.');
  }, []);

  return (
    <section className={'h-full min-h-[55rem] mt-[-1.7rem]'}>
      {contextHolder}
      <div className={'w-h-full flex-col-center'}>
        <article>
          <img className={'w-[6.7rem] h-[7.7rem]'} src={'logo.svg'} alt={'로고'} />
        </article>
        <article
          className={
            'flex-row-center border-solid border-[0.6px] border-header-gray w-[39rem] h-[23.5rem] min-h-[23.5rem] shadow-sign-up px-20 my-10'
          }
        >
          <form onSubmit={onSubmit} className={'flex-col-center justify-center w-h-full'}>
            <div
              className={
                'border-gray-light w-full flex flex-col justify-around rounded-xl shadow-sign-up-info mb-[1rem]'
              }
            >
              {/* 이메일 */}
              <span
                className={
                  'w-full h-[4.5rem] flex-row-center border-solid border-b border-gray-light'
                }
              >
                <label className={'w-h-full flex-row-center'}>
                  <span className={'w-1/6 h-full flex-row-center text-3xl '}>
                    <MdOutlineMailOutline className={'fill-gray-light'} />
                  </span>
                  <input
                    type="email"
                    value={email}
                    onChange={onChangeEmail}
                    placeholder={'이메일'}
                    required
                    maxLength={30}
                    className={'w-5/6 h-full text-xl rounded-xl'}
                  />
                </label>
              </span>

              {/*비밀번호*/}
              <span className={'w-full h-[4.5rem] flex-row-center'}>
                <label className={'w-h-full flex-row-center'}>
                  <span className={'w-1/6 h-full flex-row-center text-3xl'}>
                    <AiOutlineLock className={'fill-gray-light'} />
                  </span>
                  <span className={'w-5/6 h-full flex'}>
                    <input
                      type={'password'}
                      value={password}
                      onChange={onChangePassword}
                      placeholder={'비밀번호'}
                      maxLength={15}
                      required
                      className={`w-h-full text-xl rounded-xl ${password && 'tracking-[-0.3rem]'}`}
                    />
                  </span>
                </label>
              </span>
            </div>

            <button
              className={
                'flex-row-center border-base border-header-gray rounded-md w-[29rem] h-12 mt-10 py-7 font-bold text-xl bg-orange text-white'
              }
            >
              로그인
            </button>
          </form>
        </article>

        <article className={'w-[39rem] flex-col-center text-xl font-bold'}>
          <div
            className={'flex-row-center w-full border-solid border-b border-gray-spring pb-7 mb-5'}
          >
            <Link to={'/pwinquiry'} className={'w-[46%] text-right'}>
              <button>비밀번호 찾기</button>
            </Link>
            <div className={'w-[8%] text-center'}>
              <span className={''}>|</span>
            </div>
            <Link to={'/signup'} className={'w-[46%]'}>
              <button>회원가입</button>
            </Link>
          </div>
          <div className={'flex-col-center w-full'}>
            <button className={'mb-7'}>
              <span>간편 로그인</span>
            </button>
            <div className={'w-3/5 flex-row-center justify-evenly'}>
              <button>
                <img className={'w-14'} src={'google.svg'} alt={'google'} />
              </button>
              <button>
                <img className={'w-14'} src={'kakao.svg'} alt={'kakao'} />
              </button>
            </div>
          </div>
        </article>
      </div>
    </section>
  );
}
