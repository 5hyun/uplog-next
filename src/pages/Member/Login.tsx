import React, { FormEvent, useCallback, useEffect } from 'react';
import { MdOutlineMailOutline } from 'react-icons/md';
import useInput from '@/hooks/useInput.ts';
import { AiOutlineLock } from 'react-icons/ai';
import { Link, useNavigate } from 'react-router-dom';
import { useGoogleLogin } from '@react-oauth/google';
import { useMessage } from '@/hooks/useMessage.ts';
import { GetUserInfo, LoginInfo, SaveUserInfo } from '@/typings/member.ts';
import { useMutation } from 'react-query';
import { useRecoilState } from 'recoil';
import { loginStatus } from '@/recoil/User/atom.ts';
import { useCookies } from 'react-cookie';
import { loginAPI } from '@/api/Members/Login-Signup.ts';
import { useGetAllProduct } from '@/components/Product/hooks/useGetAllProduct.ts';

export default function Login() {
  const { showMessage, contextHolder } = useMessage();
  const [email, onChangeEmail, setEmail] = useInput('');
  const [password, onChangePassword, setPassword] = useInput('');
  const navigate = useNavigate();
  const [isLogin, setIsLogin] = useRecoilState(loginStatus);
  const [cookies, setCookie, removeCookie] = useCookies();

  const { mutate, isSuccess } = useMutation(loginAPI, {
    onSuccess: (data: GetUserInfo | string) => {
      if (typeof data === 'string') {
        showMessage('error', '아이디 또는 비밀번호를 잘못 입력하셨습니다.');
        return;
      }

      const { id, email, nickname, name, position, accessToken, refreshToken, image } = data;
      const userInfo: SaveUserInfo = {
        id,
        nickname,
        name,
        position,
        email,
        image,
      };

      console.log(image);

      sessionStorage.setItem('accessToken', accessToken);
      setCookie('refreshToken', refreshToken, { path: '/' });
      sessionStorage.setItem('userInfo', JSON.stringify(userInfo));
      setIsLogin(true);

      // const productList = useGetAllProduct();
      // console.log(productList);
      navigate('/workspace/1', { state: { isLogin: true } });
    },
    onError: () => {
      showMessage('error', '아이디 또는 비밀번호를 잘못 입력하셨습니다.');
    },
  });

  const onSubmit = useCallback(
    (e: FormEvent<HTMLFormElement>) => {
      e.preventDefault();

      if (!email || !password) {
        showMessage('warning', '이메일과 비밀번호를 입력해주세요.');
        return;
      }

      const loginInfo: LoginInfo = { email, password };
      mutate(loginInfo);
    },
    [email, password]
  );

  const login = useGoogleLogin({
    onSuccess: (codeResponse) => console.log(codeResponse),
    flow: 'auth-code',
  });

  useEffect(() => {
    if (sessionStorage.getItem('accessToken') && sessionStorage.getItem('userInfo')) navigate('/');
  }, []);

  return (
    <section className={'h-full'}>
      {contextHolder}
      <div className={'w-h-full flex-col-center'}>
        <article>
          <img className={'w-[5.7rem] h-[6.7rem]'} src={'logo.svg'} alt={'logo'} />
        </article>
        <article
          className={
            'flex-row-center border-base w-[37rem] h-[23.5rem] min-h-[23.5rem] shadow-sign-up px-[4.5rem] py-4 my-10'
          }
        >
          <form onSubmit={onSubmit} className={'flex-col-center justify-center w-h-full'}>
            <div
              className={
                'border-solid border-[0.6px] border-gray-border w-full flex flex-col justify-around rounded-xl shadow-sign-up-info mb-[1rem]'
              }
            >
              {/* 이메일 */}
              <span
                className={
                  'w-full h-[4.5rem] flex-row-center border-solid border-b border-gray-border'
                }
              >
                <label className={'w-h-full flex-row-center'}>
                  <span className={'w-1/6 h-full flex-row-center text-2xl '}>
                    <MdOutlineMailOutline
                      className={`${email ? 'fill-orange' : 'fill-gray-light'}`}
                    />
                  </span>
                  <input
                    type="text"
                    value={email}
                    onChange={onChangeEmail}
                    placeholder={'이메일'}
                    maxLength={30}
                    className={'w-5/6 h-full text-lg rounded-xl'}
                  />
                </label>
              </span>

              {/*비밀번호*/}
              <span className={'w-full h-[4.5rem] flex-row-center'}>
                <label className={'w-h-full flex-row-center'}>
                  <span className={'w-1/6 h-full flex-row-center text-2xl'}>
                    <AiOutlineLock className={`${password ? 'fill-orange' : 'fill-gray-light'}`} />
                  </span>
                  <span className={'w-5/6 h-full flex'}>
                    <input
                      type={'password'}
                      value={password}
                      onChange={onChangePassword}
                      placeholder={'비밀번호'}
                      maxLength={15}
                      className={`w-h-full text-lg rounded-xl`}
                    />
                  </span>
                </label>
              </span>
            </div>

            <button
              className={
                'flex-row-center rounded-md w-[28rem] h-12 mt-10 py-7 font-bold text-xl bg-orange text-white'
              }
            >
              로그인
            </button>
          </form>
        </article>

        <article className={'w-[37rem] flex-col-center text-lg font-bold'}>
          <nav
            // className={'flex-row-center w-full border-solid border-b border-gray-spring pb-7 mb-5'}
            className={'flex-row-center w-full pb-7 mb-5'}
          >
            <Link to={'/pwinquiry'} className={'w-[46%] text-right'}>
              <button>비밀번호 찾기</button>
            </Link>
            <div className={'w-[8%] text-center'}>
              <span className={''}>|</span>
            </div>
            <Link to={'/signup'} className={'w-[46%]'}>
              <button>회원가입</button>
            </Link>
          </nav>
          {/*<div className={'flex-col-center w-full'}>*/}
          {/*  <span className={'mb-7 text-lg font-bold'}>간편 로그인</span>*/}
          {/*  <div className={'w-3/5 flex-row-center justify-evenly'}>*/}
          {/*    <button onClick={() => login()}>*/}
          {/*      <img className={'w-12'} src={'google.svg'} alt={'google'} />*/}
          {/*    </button>*/}
          {/*    <button>*/}
          {/*      <img className={'w-12'} src={'kakao.svg'} alt={'kakao'} />*/}
          {/*    </button>*/}
          {/*  </div>*/}
          {/*</div>*/}
        </article>
      </div>
    </section>
  );
}
