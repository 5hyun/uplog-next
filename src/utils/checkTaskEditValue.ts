import { TaskData } from '@/typings/task.ts';

export function checkTaskEditValue(editTask: TaskData): boolean {
  if (editTask.taskName === '') {
    return false;
  }
  if (editTask.targetMemberInfoDTO && editTask.targetMemberInfoDTO.id === 0) {
    return false;
  }
  if (editTask.menuId === 0) {
    return false;
  }
  if (editTask.teamId === 0) {
    return false;
  }
  if (editTask.startTime === '' || editTask.endTime === '') {
    return false;
  }

  return true;
}
