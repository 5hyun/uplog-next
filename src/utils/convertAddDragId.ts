import { StatusTaskData, Tasks } from '@/typings/task.ts';
import { useState } from 'react';

export function convertAddDragId(tasks: StatusTaskData): Tasks {
  const [convertedTasks, setConvertedTasks] = useState<Tasks>({
    PROGRESS_BEFORE: [],
    PROGRESS_IN: [],
    PROGRESS_COMPLETE: [],
  });

  const beforeStatusTasks = tasks.PROGRESS_BEFORE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_BEFORE: beforeStatusTasks,
  });

  const inStatusTasks = tasks.PROGRESS_IN.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_IN: inStatusTasks,
  });

  const completeStatusTasks = tasks.PROGRESS_COMPLETE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_COMPLETE: completeStatusTasks,
  });

  return convertedTasks;
}
