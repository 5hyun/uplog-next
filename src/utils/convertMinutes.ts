// ** 타이머 분:초 구하는 함수 */
export function convertMinutes(seconds: number): string {
  const minutes = Math.floor(seconds / 60);
  const remainingSeconds = seconds % 60;
  const formattedSeconds = remainingSeconds < 10 ? `0${remainingSeconds}` : `${remainingSeconds}`;

  if (minutes === 0) {
    return `00:${formattedSeconds}`;
  }

  return `${minutes}:${formattedSeconds}`;
}
