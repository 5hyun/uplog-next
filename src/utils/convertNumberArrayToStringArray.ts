export function convertNumberArrayToStringArray(numbers: number[]): string[] {
  // 배열의 각 요소를 문자열로 변환하여 새로운 배열을 생성합니다.
  const strings: string[] = numbers.map((number) => number.toString());
  return strings;
}
