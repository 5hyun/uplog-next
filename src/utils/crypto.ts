import CryptoJS from 'crypto-js';

/* Salt 비밀 키 */
// const salt = import.meta.env.VITE_SALT;
const salt = '1234';

/* 암호화 */
export const encrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';
  return CryptoJS.AES.encrypt(text, salt).toString();
};

/* 복호화 */
export const decrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';

  try {
    const bytes = CryptoJS.AES.decrypt(text, salt); // 복호화 시도
    return bytes.toString(CryptoJS.enc.Utf8);
  } catch (error) {
    return ''; // 에러 발생 시 빈 문자열 반환
  }
};
