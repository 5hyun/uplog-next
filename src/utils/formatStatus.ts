// ** 진행 전,중,후 상태로 변환하기 위한 함수 */

export function formatStatus(status: string): string {
  if (status === 'PROGRESS_BEFORE') {
    return '진행 전';
  } else if (status === 'PROGRESS_IN') {
    return '진행 중';
  }

  return '진행 후';
}
