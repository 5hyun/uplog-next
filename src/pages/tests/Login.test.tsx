import { describe, expect, test } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import Login from '../Login';
import '@testing-library/jest-dom';
import { withRouter } from '../../tests/utils';
import { Route } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { GoogleOAuthProvider } from '@react-oauth/google';

describe('Login', () => {
  // const clientId = import.meta.env.VITE_GOOGLE_CLIENTID;

  test('login correctly', () => {
    const component = renderer.create(withRouter(<Route path={'/login'} element={<Login />} />));

    expect(component.toJSON()).toMatchSnapshot();
  });

  // test('renders the login component correctly', () => {
  //   render(
  //     withRouter(
  //       <GoogleOAuthProvider clientId={clientId}>
  //         <Route path={'/'} element={<Login />} />
  //       </GoogleOAuthProvider>
  //     )
  //   );
  //   // Check if the email and password input fields are rendered correctly
  //   const emailInput = screen.getByPlaceholderText('이메일');
  //   const passwordInput = screen.getByPlaceholderText('비밀번호');
  //   // expect(emailInput).toBeInTheDocument();
  //   // expect(passwordInput).toBeInTheDocument();
  //
  //   // // Check if the 로그인 button is rendered correctly
  //   // const loginButton = getByText('로그인');
  //   // expect(loginButton).toBeInTheDocument();
  // });
});
