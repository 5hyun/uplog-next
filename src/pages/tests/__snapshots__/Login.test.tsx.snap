// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login login correctly 1`] = `
<section
  className="h-full"
>
  <div
    className="w-h-full flex-col-center"
  >
    <article>
      <img
        alt="로고"
        className="w-[5.7rem] h-[6.7rem]"
        src="logo.svg"
      />
    </article>
    <article
      className="flex-row-center border-base w-[39rem] h-[23.5rem] min-h-[23.5rem] shadow-sign-up px-20 my-10"
    >
      <form
        className="flex-col-center justify-center w-h-full"
        onSubmit={[Function]}
      >
        <div
          className="border-solid border-[0.6px] border-gray-border w-full flex flex-col justify-around rounded-xl shadow-sign-up-info mb-[1rem]"
        >
          <span
            className="w-full h-[4.5rem] flex-row-center border-solid border-b border-gray-border"
          >
            <label
              className="w-h-full flex-row-center"
            >
              <span
                className="w-1/6 h-full flex-row-center text-3xl "
              >
                <svg
                  className="fill-gray-light"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                  />
                </svg>
              </span>
              <input
                className="w-5/6 h-full text-xl rounded-xl"
                maxLength={30}
                onChange={[Function]}
                placeholder="이메일"
                required={true}
                type="email"
                value=""
              />
            </label>
          </span>
          <span
            className="w-full h-[4.5rem] flex-row-center"
          >
            <label
              className="w-h-full flex-row-center"
            >
              <span
                className="w-1/6 h-full flex-row-center text-3xl"
              >
                <svg
                  className="fill-gray-light"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                  />
                </svg>
              </span>
              <span
                className="w-5/6 h-full flex"
              >
                <input
                  className="w-h-full text-xl rounded-xl "
                  maxLength={15}
                  onChange={[Function]}
                  placeholder="비밀번호"
                  required={true}
                  type="password"
                  value=""
                />
              </span>
            </label>
          </span>
        </div>
        <button
          className="flex-row-center rounded-md w-[29rem] h-12 mt-10 py-7 font-bold text-xl bg-orange text-white"
        >
          로그인
        </button>
      </form>
    </article>
    <article
      className="w-[39rem] flex-col-center text-xl font-bold"
    >
      <nav
        className="flex-row-center w-full border-solid border-b border-gray-spring pb-7 mb-5"
      >
        <a
          className="w-[46%] text-right"
          href="/pwinquiry"
          onClick={[Function]}
        >
          <button>
            비밀번호 찾기
          </button>
        </a>
        <div
          className="w-[8%] text-center"
        >
          <span
            className=""
          >
            |
          </span>
        </div>
        <a
          className="w-[46%]"
          href="/signup"
          onClick={[Function]}
        >
          <button>
            회원가입
          </button>
        </a>
      </nav>
      <div
        className="flex-col-center w-full"
      >
        <span
          className="mb-7 text-xl font-bold"
        >
          간편 로그인
        </span>
        <div
          className="w-3/5 flex-row-center justify-evenly"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="google"
              className="w-14"
              src="google.svg"
            />
          </button>
          <button>
            <img
              alt="kakao"
              className="w-14"
              src="kakao.svg"
            />
          </button>
        </div>
      </div>
    </article>
  </div>
</section>
`;
